tests/test_gpu_examples.py

```diff
@@ -1878,6 +1879,184 @@ def test_causal_lm_training_aqlm(self):
             assert trainer.state.log_history[-1]["train_loss"] is not None
 
 
+@require_torch_gpu
+@require_hqq
+@unittest.skipUnless(
+    version.parse(importlib.metadata.version("transformers")) >= version.parse("4.36.1"),
+    "test requires `transformers>=4.36.1`",
+)
+class PeftHqqGPUTests(unittest.TestCase):
+    r"""
+    HQQ + peft tests
+    """
+
+    def setUp(self):
+        self.causal_lm_model_id = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.causal_lm_model_id)
+
+    def tearDown(self):
+        r"""
+        Efficient mechanism to free GPU memory after each test. Based on
+        https://github.com/huggingface/transformers/issues/21094
+        """
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    @pytest.mark.single_gpu_tests
+    @parameterized.expand([False, True])
+    def test_causal_lm_training_hqq(self, use_dora):
+        r"""
+        Test the CausalLM training on a single GPU device. The test would simply fail if the adapters are not set
+        correctly.
+        """
+
+        from hqq.core.quantize import BaseQuantizeConfig
+        from hqq.engine.hf import HQQModelForCausalLM
```

Comment:
Could you adapt this test in the future once https://github.com/huggingface/transformers/pull/29637 gets merged so that it leverages `AutoModelForCausalLM` ?

Reply:
I'm confused what should I do here. Could you please elaborate?